### PR TITLE
Replace @windowsxp_only macro with runtime check

### LINKED ISSUE #11108: windows cross-compile error in symlink function definition
the windowsxp-only test should be at runtime, not compile-time. otherwise a wine-cross compile doesn't work on my windows 8 test machine:
```
     * file                exception on 1: ERROR: LoadError: WindowsXP does not support soft symlinks
 in symlink at fs.jl:168
 in runtests at J:\julia32\test\testdefs.jl:77
 in anonymous at multi.jl:626
 in run_work_thunk at multi.jl:587
 in remotecall_fetch at multi.jl:675
 in anonymous at task.jl:1390
while loading J:\julia32\test\file.jl, in expression starting on line 17
```

### PR DESCRIPTION
**Pull request by tkelman**
fixes #11108

### CODE REVIEW

File: base/osutils.jl [hunk 1]
```diff
@@ -56,11 +56,3 @@ end
 @unix_only windows_version() = (0,0)
 
 WINDOWS_XP_VER = (5,1)
```
Comment (vtjnash) [inline base/osutils.jl]: (5,2) is also a Windows XP version (Vista started with  6.0)
https://msdn.microsoft.com/en-us/library/windows/desktop/ms724832%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396

seems like a pre-existing bug, but maybe better to compare against WINDOWS_VISTA_VER?

Comment (tkelman) [inline base/osutils.jl]: Oh, no kidding, guess my wine thinks it's either server 2003 or xp 64. I don't have either of those handy aside from in Wine, so I think comparing to vista does make sense. Will change that.

**Comment by tkelman** [comment]
Merging. @vtjnash I trust you'll open a new issue and/or let me know if there are any other problems with this.

**Comment by vtjnash** [comment]
looked good. i was assuming you would hit merge when travis finished.

**Pull request merged**